src/MinimalDelegation.sol

```diff
@@ -35,4 +48,20 @@ contract MinimalDelegation {
         delete MinimalDelegationStorageLib.get().keyStorage[keyHash];
         emit Revoked(keyHash);
     }
+
+    function supportsExecutionMode(bytes32 mode) external pure override returns (bool result) {
+        return mode.isBatchedCall();
+    }
+
+    function _authorizeCaller() private view {
+        if (msg.sender != address(this)) revert IERC7821.Unauthorized();
+    }
+
+    function _execute(Calls[] memory calls) private {
+        for (uint256 i = 0; i < calls.length; i++) {
+            Calls memory _call = calls[i];
```

Comment:
per the spec, we should cast address(0) to address(this) here for calldata gas savings